### DISCUSSION TITLE
Confused on how to Integrate

### DISCUSSION THREAD

**Discussion by mahyarmirrashed** [nix-community/haumea#53] [General]
I've read through the documentation a few times and tried to use a GPT to make sense of it, but I've still had some issues figuring out how to integrate this into my project. It would help me, as a Nix newbie, to have the backstory and a guide/tutorial in integrating Haumea into my existing `nix-config`.

**Comment by mahyarmirrashed** [comment]
After much more tinkering, and with the help of GPTo1, I was able to get it integrated into my project. Going to definitely include this in my tutorial series. Nifty tool!

**Reply by tahlonbrahic** [reply]
I have been confused on haumea for months yet I think it is a pattern that fits well with Nix so I have been constantly trying to integrate it into my projects. Your flake was the first working example I have seen in the wild. Thank you for finally letting me see some examples!

**Comment by mahyarmirrashed** [comment]
Coming back to this, I think the biggest improvement desperately needed for this @figsoda is some examples of its usage. The documentation is all fine and dandy, but it rushes over my head without any concrete examples to build from. I tried my best looking at other people's examples here on GitHub, to no avail.

For example, I'm pretty sure that it should be possible to hoist in `png` files with Haumea so that I can reference `./wallpapers/gruvbox/breeze.png` as `wallpapers.gruvbox.breeze`. Yet, the documentation isn't obvious on this.

**Reply by tahlonbrahic** [reply]
I agree, it was easier to integrate by looking at the source then by the documentation when there are not examples.
